app/scripts/controllers/buildConfig.js

```diff
@@ -197,6 +203,44 @@ angular.module('openshiftConsole')
           }
         );
 
+        DataService.list("configmaps", context, null, { errorNotification: false }).then(function(configMapData) {
+          configMapDataOrdered = orderByDisplayName(configMapData.by("metadata.name"));
+          $scope.valueFromObjects = configMapDataOrdered.concat(secretDataOrdered);
+        }, function(e) {
+          if (e.code === 403) {
+            return;
+          }
+
+          NotificationsService.addNotification({
+            id: "create-builder-list-config-maps-error",
+            type: "error",
+            message: "Could not load config maps.",
+            details: getErrorDetails(e)
+          });
+        });
+
+        DataService.list("secrets", context, null, { errorNotification: false }).then(function(secretData) {
+          secretDataOrdered = orderByDisplayName(secretData.by("metadata.name"));
+          $scope.valueFromObjects = secretDataOrdered.concat(configMapDataOrdered);
+          var secretsByType = SecretsService.groupSecretsByType(secretData);
+          var secretNamesByType =_.mapValues(secretsByType, function(secretData) {return _.map(secretData, 'metadata.name');});
+          // Add empty option to the image/source secrets
+          $scope.secretsByType = _.each(secretNamesByType, function(secretsArray) {
+            secretsArray.unshift("");
+          });
```

Comment:
I think lines 225-230 can be deleted. I don't believe we do anything with `secretsByType` on this page.